testing/python/fixture.py

```diff
@@ -490,6 +490,21 @@ def test_func(something): pass
         print(ss.stack)
         assert teardownlist == [1]
 
+    def test_mark_as_fixture_with_prefix_and_decorator_fails(self, testdir):
+        testdir.makeconftest("""
+            import pytest
+
+            @pytest.fixture
+            def pytest_funcarg__marked_with_prefix_and_decorator():
+                pass
+        """)
+        result = testdir.runpytest_subprocess()
+        assert result.ret != 0
+        result.stdout.fnmatch_lines([
+            "*AssertionError: *",
+            "*pytest_funcarg__marked_with_prefix_and_decorator*"
```

Comment:
To be honest we should have a better message than just the function name... perhaps:

```
pytest_funcarg__marked_with_prefix_and_decorator is also decorated with @pytest.fixture; 
It is recommended to remove the "pytest_funcarg__" prefix.
```